File: LuaLibs/src/MyLua/Libs/Table/TableItemHeader.lua
require "MyLua.Libs.Core.GlobalNS"

local M = GlobalNS.Class()
M.clsName = "TableItemHeader"
GlobalNS[M.clsName] = M

function M:ctor()
    self.m_uID = 0;              -- 唯一 ID
    self.m_offset = 0;           -- 这一项在文件中的偏移
end

-- 解析头部
function M:parseHeaderByteBuffer(bytes)
    bytes.readUnsignedInt32(m_uID);
    bytes.readUnsignedInt32(m_offset);
end

return M

File: JSLib/src/Table/TableItemHeader.js
namespace SDK.Lib
{
    public class TableItemHeader
    {
        public uint m_uID;              // 唯一 ID
        public uint m_offset;           // 这一项在文件中的偏移

        // 解析头部
        virtual public void parseHeaderByteBuffer(ByteBuffer bytes)
        {
            bytes.readUnsignedInt32(ref m_uID);
            bytes.readUnsignedInt32(ref m_offset);
        }
    }
}